src/net/skbuff/skb_data.c

```diff
@@ -125,9 +139,17 @@ void skb_data_free(struct sk_buff_data *skb_data) {
 	sp = ipl_save();
 	{
 		if (--skb_data->links == 0) {
-			assert(pool_belong(&skb_data_pool, skb_data));
-
-			pool_free(&skb_data_pool, skb_data);
+			switch (skb_data->alloc_type) {
+			case ALLOCATED_POOL:
```

Comment:
Another option is to use `pool_belong`, in this case no need to store allocation metainfo

Reply:
Yes, that's exactly like it was previously done, but as for me, it is more safely and clearly to state explicitly how the object is allocated.